synapse/cores/common.py

```diff
@@ -3749,3 +3759,78 @@ def delBlobValu(self, key):
             NoSuchName: If the key is not present in the store.
         '''
         return self.store.delBlobValu(key)
+
+    def _onSetMembranes(self, membranes):
+        [self._initMembrane(name, rules) for name, rules in membranes]
+
+    def _initMembrane(self, name, rules):
+        if self._core_membranes.get(name):
```

Comment:
maybe we can check if rules are updated here and update the rules?

Reply:
makes sense, adding the change now

Reply:
fixed